### PR TITLE
Add new portal storm monster swarm structure

### PR DESCRIPTION
**Pull request by Ramza13**
#### Summary
None

#### Purpose of change
Part of #52475
New monster
Remove old nether monster group spawn, so no more flaming eyes, krecks, etc.  This should mean no more nether creatures smelling you and breaking into vehicles/shelters. 

#### Describe the solution
Swarm structure's have no attack and can't move.  All they can do is either teleport closer to you or summon a swarm of twisted blades which cause pain and stamina loss but no damage.
Fix location_variable handling for npc_vars

### CODE REVIEW

File: data/json/monsters/nether.json [hunk 1]
```diff
@@ -1093,5 +1093,61 @@
...
+    "armor_bullet": 5,
+    "special_attacks": [
+      {
+        "type": "spell",
+        "spell_data": { "id": "mon_swarm_structure_attack", "min_level": 1 },
+        "cooldown": 6,
+        "monster_message": "The integrity of spacetime is violated."
+      }
+    ],
+    "death_function": { "corpse_type": "NO_CORPSE", "message": "You memories can't decide whether the %s was even there in the first place." },
```
Comment (schoolcf) [inline data/json/monsters/nether.json:1123]: rrrrr
```suggestion
    "death_function": { "corpse_type": "NO_CORPSE", "message": "Your memory can't decide whether the %s was even there in the first place." },
```

File: data/json/monsters/nether.json [hunk 2]
```diff
@@ -1093,5 +1093,61 @@
...
+    "dodge": 8,
+    "special_attacks": [
+      {
+        "type": "spell",
+        "spell_data": { "id": "mon_twisting_blade_attack", "min_level": 1 },
+        "cooldown": 1,
+        "monster_message": "You are cut in ways worse than physical."
+      }
+    ],
+    "death_function": { "corpse_type": "NO_CORPSE", "message": "You memories can't decide whether the %s was even there in the first place." },
```
Comment (schoolcf) [inline data/json/monsters/nether.json:1150]: rrrrrrrrrr
```suggestion
    "death_function": { "corpse_type": "NO_CORPSE", "message": "Your memory can't decide whether the %s was even there in the first place." },
```

**Comment by LyleSY** [comment]
Am I right this is just pulling off the wilderness spawn list for whatever terrain type it's on?

**Comment by Ramza13** [comment]
> Am I right this is just pulling off the wilderness spawn list for whatever terrain type it's on?

I don't follow at all?  This should be unrelated to anything other than portal storms which don't care about terrain.

**Pull request merged**